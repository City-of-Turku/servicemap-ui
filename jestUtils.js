import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });