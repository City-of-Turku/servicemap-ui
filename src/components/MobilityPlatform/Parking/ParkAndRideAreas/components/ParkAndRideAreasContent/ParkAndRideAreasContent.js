import { Typography } from '@mui/material';
import PropTypes from 'prop-types';
import React from 'react';
import { useIntl } from 'react-intl';
import { StyledContainer, StyledHeaderContainer, StyledTextContainer } from '../../../../styled/styled';

const ParkAndRideAreasContent = ({ parkAndRideArea, parkingStatistics }) => {
  const intl = useIntl();

  let freeParkingSpaces = 0;
  const parkingSpaceStats = parkingStatistics?.filter(item => item.id === parkAndRideArea?.id);

  const renderText = (isTitle, translationId, text) => (isTitle ? (
    <StyledHeaderContainer>
      <Typography variant="subtitle1" component="h3">
        {intl.formatMessage({
          id: translationId,
        })}
      </Typography>
    </StyledHeaderContainer>
  ) : (
    <StyledTextContainer>
      <Typography variant="body2" component="p">
        {intl.formatMessage({
          id: translationId,
        })}
        :
        {' '}
        {text}
      </Typography>
    </StyledTextContainer>
  ));

  const renderParkingCount = (capacity, parkingCount) => {
    freeParkingSpaces = capacity - parkingCount;

    return (
      <StyledTextContainer key={capacity}>
        {freeParkingSpaces > 0 ? (
          <Typography variant="body2">
            {intl.formatMessage(
              { id: 'mobilityPlatform.content.parkingSpaces.parkingCount' },
              { value: freeParkingSpaces, capacity },
            )}
          </Typography>
        ) : (
          <Typography variant="body2">
            {intl.formatMessage({ id: 'mobilityPlatform.content.parkingSpaces.empty' })}
          </Typography>
        )}
      </StyledTextContainer>
    );
  };

  const renderBusNumbers = data => data?.map(item => (
    <StyledTextContainer key={item}>
      <Typography variant="body2" component="p">
        {intl.formatMessage({ id: 'mobilityPlatform.content.parkAndRide.bus' }, { value: item })}
      </Typography>
    </StyledTextContainer>
  ));

  return (
    <StyledContainer>
      {renderText(true, 'mobilityPlatform.content.parkAndRide.title')}
      {renderText(false, 'mobilityPlatform.content.parkingSpaces.capacity', parkAndRideArea?.properties?.capacity_estimate)}
      {parkingSpaceStats?.length > 0
        && parkingSpaceStats.map(parking => renderParkingCount(parkAndRideArea?.properties?.capacity_estimate, parking?.current_parking_count))}
      {renderText(false, 'mobilityPlatform.content.parkAndRide.busNumbers')}
      {renderBusNumbers(parkAndRideArea?.properties?.bus_stop_numbers)}
    </StyledContainer>
  );
};

ParkAndRideAreasContent.propTypes = {
  parkAndRideArea: PropTypes.shape({
    id: PropTypes.string,
    properties: PropTypes.shape({
      capacity_estimate: PropTypes.number,
      bus_stop_numbers: PropTypes.arrayOf(PropTypes.number),
    }),
  }),
  parkingStatistics: PropTypes.arrayOf(
    PropTypes.shape({
      current_parking_count: PropTypes.number,
    }),
  ),
};

ParkAndRideAreasContent.defaultProps = {
  parkAndRideArea: {},
  parkingStatistics: [],
};

export default ParkAndRideAreasContent;
