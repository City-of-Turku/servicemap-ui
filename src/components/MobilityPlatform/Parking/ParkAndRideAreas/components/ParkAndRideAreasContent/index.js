import ParkAndRideAreasContent from './ParkAndRideAreasContent';

export default ParkAndRideAreasContent;
