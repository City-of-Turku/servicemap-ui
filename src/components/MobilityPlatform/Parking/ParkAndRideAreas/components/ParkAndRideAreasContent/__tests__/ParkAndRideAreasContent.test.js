// Link.react.test.js
import React from 'react';
import ParkAndRideAreasContent from '../index';
import { getRenderWithProviders } from '../../../../../../../../jestUtils';
import finnishTranslations from '../../../../../../../i18n/fi';

const mockProps = {
  parkAndRideArea: {
    id: '123abc',
    properties: {
      capacity_estimate: 4,
      bus_stop_numbers: [
        50,
        60,
      ],
    },
  },
  parkingStatistics: [
    {
      id: '123abc',
      current_parking_count: 0,
    },
  ],
};

const renderWithProviders = getRenderWithProviders({});

describe('<ParkAndRideAreasContent />', () => {
  it('should work', () => {
    const { container } = renderWithProviders(<ParkAndRideAreasContent {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<ParkAndRideAreasContent {...mockProps} />);

    const h3 = container.querySelectorAll('h3');
    const p = container.querySelectorAll('p');
    expect(h3[0].textContent).toContain(finnishTranslations['mobilityPlatform.content.parkAndRide.title']);
    expect(p[0].textContent).toContain(
      `${finnishTranslations['mobilityPlatform.content.parkingSpaces.capacity']}: ${mockProps.parkAndRideArea.properties.capacity_estimate}`,
    );
    expect(p[1].textContent).toContain(`Vapaana olevat parkkipaikat: 4 / ${mockProps.parkAndRideArea.properties.capacity_estimate}`);
    expect(p[2].textContent).toContain(finnishTranslations['mobilityPlatform.content.parkAndRide.busNumbers']);
    expect(p[3].textContent).toContain(`${finnishTranslations['mobilityPlatform.content.parkAndRide.bus'].replace(
      '{value}',
      `${mockProps.parkAndRideArea.properties.bus_stop_numbers[0]}`,
    )}`);
    expect(p[4].textContent).toContain(`${finnishTranslations['mobilityPlatform.content.parkAndRide.bus'].replace(
      '{value}',
      `${mockProps.parkAndRideArea.properties.bus_stop_numbers[1]}`,
    )}`);
  });
});
