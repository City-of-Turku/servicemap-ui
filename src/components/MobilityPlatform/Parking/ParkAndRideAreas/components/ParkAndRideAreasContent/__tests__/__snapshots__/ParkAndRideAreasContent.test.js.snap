// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ParkAndRideAreasContent /> should work 1`] = `
<div>
  <div
    class="css-1ynyhby"
  >
    <div
      class="css-1dznbe2"
    >
      <h3
        class="MuiTypography-root MuiTypography-subtitle1 css-dlbwt1-MuiTypography-root"
      >
        Liityntäpysäköintipaikka
      </h3>
    </div>
    <div
      class="css-1fhgjcy"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
      >
        Parkkipaikkojen määrä
        :
         
        4
      </p>
    </div>
    <div
      class="css-1fhgjcy"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
      >
        Vapaana olevat parkkipaikat: 4 / 4
      </p>
    </div>
    <div
      class="css-1fhgjcy"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
      >
        Läheltä kulkevat linja-autot
        :
         
      </p>
    </div>
    <div
      class="css-1fhgjcy"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
      >
        Linja: 50
      </p>
    </div>
    <div
      class="css-1fhgjcy"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
      >
        Linja: 60
      </p>
    </div>
  </div>
</div>
`;
